ecr-login/ecr_test.go

```diff
@@ -118,3 +119,83 @@ func TestListFailure(t *testing.T) {
 	assert.Error(t, err)
 	assert.Len(t, serverList, 0)
 }
+
+func TestAddIgnored(t *testing.T) {
+	factory := &mock_api.MockClientFactory{}
+
+	helper := NewECRHelper(WithClientFactory(factory))
+
+	os.Setenv("AWS_ECR_IGNORE_CREDS_STORAGE", "true")
+	err := helper.Add(&credentials.Credentials{
+		ServerURL: proxyEndpoint,
+		Username:  "AWS",
+		Secret:    "supersecret",
+	})
+
+	assert.Nil(t, err)
+}
+
+func TestAddNotImplemented(t *testing.T) {
+	tests := []struct {
+		name   string
+		setEnv func()
+	}{
+		{"unset", func() { os.Unsetenv("AWS_ECR_IGNORE_CREDS_STORAGE") }},
+		{"false", func() { os.Setenv("AWS_ECR_IGNORE_CREDS_STORAGE", "false") }},
+		{"0", func() { os.Setenv("AWS_ECR_IGNORE_CREDS_STORAGE", "0") }},
+		{"empty string", func() { os.Setenv("AWS_ECR_IGNORE_CREDS_STORAGE", "") }},
```

Comment:
_Consider_ using `t.Setenv()` to set environment variables, and add a helper function like the following to unset them.

These make it less likely that environment variables (globals) don't interfere with other tests, and `t.Setenv` will signal if it's being run in a parallel context with `t.Parallel`.

```go
// unsetEnv unsets an environment variable and registers a cleanup function that
// restores it after the test is complete. Note: tests that modify environment
// variables may not be run in parallel.
//
// See also [testing.T.Setenv]
func unsetEnv(t *testing.T, key string) {
	t.Helper()

	// save original value
	originalValue, wasSet := os.LookupEnv(key)

	// unset the environment variable
	os.Unsetenv(key)

	// restore the original value if necessary
	if wasSet {
		t.Cleanup(func() {
			os.Setenv(key, originalValue)
		})
	}
}
```

Reply:
Implemented. Thanks for this!